packages/core/entities-config-editor/src/types/fields.ts

```diff
@@ -0,0 +1,143 @@
+export const FIELD_TYPES = {
+  STRING: 'string', // Input, Textarea, Select
+  NUMBER: 'number', // Input
+  INTEGER: 'integer', // Input
+  BOOLEAN: 'boolean', // Checkbox
+  FOREIGN: 'foreign',
+  ARRAY: 'array', // Array
+  SET: 'set', // Array
+  MAP: 'map', // Map
+  RECORD: 'record', // Record
+  FUNCTION: 'function', // not used
+  JSON: 'json', // not used
+} as const
+
+export type FieldType = typeof FIELD_TYPES[keyof typeof FIELD_TYPES]
+
+export interface FieldProps<V = any, S extends FieldSchema = FieldSchema> {
+  name?: string
+  schema: S
+  value?: V
+
+  /** DEBUGGING: An stack that contains the information of the parent fields */
+  parentStack?: string[]
+}
+
+export interface FieldEmits<V = any> {
+  (e: 'update-value', value: V): void
+}
+
+export type AtLeastOneOfEntityCheck = { at_least_one_of: string[] }
+
+export type EntityCheck = AtLeastOneOfEntityCheck
+
+export const isAtLeastOneOfEntityCheck = (check: EntityCheck): check is AtLeastOneOfEntityCheck =>
+  Object.prototype.hasOwnProperty.call(check, 'at_least_one_of')
+
+export interface FieldSchema {
+  type: FieldType
+  required?: boolean
+  default?: any
+  description?: string
+  referenceable?: boolean
+
+  one_of?: any[]
+
+  help?: string
+
+  entity_checks?: EntityCheck[]
+}
+
+export type NamedFieldSchema = { [name: string]: FieldSchema }
+
+export interface StringFieldSchema extends FieldSchema {
+  type: typeof FIELD_TYPES.STRING
+
+  one_of?: string[]
+
+  len_eq?: number
+  len_min?: number
+  len_max?: number
+
+  match?: string
+
+  match_none?: {
+    pattern: string
+    err: string
+  }[]
+
+  match_all?: {
+    pattern: string
+    err: string
+  }[]
+
+  match_any?: {
+    patterns: string[]
+    err: string
+  }
+
+  pattern?: string
+}
+
+export const isStringField = (schema: FieldSchema): schema is StringFieldSchema =>
+  schema.type === FIELD_TYPES.STRING
+
+export interface NumberLikeFieldSchema extends FieldSchema {
+  type: typeof FIELD_TYPES.NUMBER | typeof FIELD_TYPES.INTEGER
+
+  one_of?: number[]
+
+  between?: [min: number, max: number]
+  gt?: number
+}
+
+export const isNumberLikeField = (schema: FieldSchema): schema is NumberLikeFieldSchema =>
+  schema.type === FIELD_TYPES.NUMBER || schema.type === FIELD_TYPES.INTEGER
```

Comment:
Personally I'd prefer

```suggestion
  [FIELD_TYPES.NUMBER, FIELD_TYPES].includes(schema.type)
```